logictest: skip `alter_primary_key` under race

yuzefovich opened the pull request:
We tried to fix this by disabling metamorphic batch sizes, but we just saw another timeout even with that patch applied, so let's just skip this test under race.

Fixes: #139636.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/139659)
<!-- Reviewable:end -->

DrewKimball reviewed (commented):
> Reviewed 15 of 15 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/139659)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @michae2 and @yuzefovich)

___
*[`pkg/sql/logictest/testdata/logic_test/alter_primary_key` line 1 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/139659#-OHJAQs-3TTD9yqamjUq:-OHJAQs-3TTD9yqamjUr:b-ynf8kw) ([raw file](https://github.com/cockroachdb/cockroach/blob/c4de0a98297878922751ba78bdee2a4e6a2c59b6/pkg/sql/logictest/testdata/logic_test/alter_primary_key#L1)):*
> ```
> # LogicTest: !metamorphic-batch-sizes
> ```

Don't we need to add `skip under race`?


<!-- Sent from Reviewable.io -->

DrewKimball reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/139659#-:-OHJAhTp-FFNN3hARcNX:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @michae2 and @yuzefovich)

___
*[`pkg/sql/logictest/testdata/logic_test/alter_primary_key` line 1 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/139659#-OHJAQs-3TTD9yqamjUq:-OHJAiab4cqwdg2C3lkh:bxjidwv) ([raw file](https://github.com/cockroachdb/cockroach/blob/c4de0a98297878922751ba78bdee2a4e6a2c59b6/pkg/sql/logictest/testdata/logic_test/alter_primary_key#L1)):*
<details><summary><i>Previously, DrewKimball (Drew Kimball) wrote…</i></summary><blockquote>

Don't we need to add `skip under race`?
</blockquote></details>

Nevermind - the change is hidden in reviewable for some reason.


<!-- Sent from Reviewable.io -->

yuzefovich committed:
logictest: skip `alter_primary_key` under race

We tried to fix this by disabling metamorphic batch sizes, but we just
saw another timeout even with that patch applied, so let's just skip
this test under race.

Release note: None

yuzefovich reviewed (commented):
TFTR!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/139659#-:-OHLTRiN5zmugIpXaOv0:b-oduykh)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @DrewKimball and @michae2)

___
*[`pkg/sql/logictest/testdata/logic_test/alter_primary_key` line 1 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/139659#-OHJAQs-3TTD9yqamjUq:-OHLTCFO3kVin_n0oCha:b-r5yaku) ([raw file](https://github.com/cockroachdb/cockroach/blob/c4de0a98297878922751ba78bdee2a4e6a2c59b6/pkg/sql/logictest/testdata/logic_test/alter_primary_key#L1)):*
<details><summary><i>Previously, DrewKimball (Drew Kimball) wrote…</i></summary><blockquote>

Nevermind - the change is hidden in reviewable for some reason.
</blockquote></details>

TIL about `skip under race` - I like it more than modifying the templates. The only difference is that it doesn't support skipping under specific configs, which seems ok in this case, so I switched the change to use `skip under race`.


<!-- Sent from Reviewable.io -->
